### ISSUE TITLE
Review and Commenting Email notifications sender

### ISSUE THREAD

**Issue by jmockler** [maqetta/maqetta#437]
Problem: currently the sender is 'noreply' for email notifications. Which could be seen as spam for some email programs and so automatically removed from a requested reviewers inbox.

Solution: change the send to show the Name or email of the person who sent the Review Invitation.

@jmockler @JonFerraiolo @EldonReturn
 Review and Commenting Review - May 12, 2011

**Comment by peller** [comment]
well, it's true, you can't reply to it.  I'm not sure we can fake a return address without setting off even more spam alarms

**Comment by JonFerraiolo** [comment]
You sure you can't reply to it? The proposal is that we set reply-to field in the header to the author's email address. Can't we do that?

**Comment by peller** [comment]
I don't think you can just stick an e-mail address in the from field.  That will look like spam.  I'm suggesting that unless we make it legit, we should leave something like 'noreply'

**Comment by jmockler** [comment]
Shouldn't a user be able to reply to these notifications? What if the Review requester didn't add enough info to the description? 

We include the Senders email in the body of the notifications.. why not identify them as the sender and enable reply to?

**Comment by jmockler** [comment]
Why can't we identify the Sender of the review as the person who sent the email. That seems more informative to me than the 'noreply' moniker. And would be more useful to the participants in the review.

Whether its the users email or their Maqetta name doesn't really matter. Just not the 'noreply'.

**Comment by peller** [comment]
It's a technical issue.  I don't doubt that it would be nice to have an e-mail come from the user that one could reply to.  We need to investigate to see what's required to do this.

**Comment by jmockler** [comment]
Understood @peller.

**Comment by peller** [comment]
Sorry to beat a dead horse, but I missed the last line of your comment.  Unless you can really reply to the e-mail, I don't think it should look like it's actually from a person.  Perhaps something indicating that it's from a Maqetta server would be friendlier than just no reply, but using something with no-reply in the e-mail address is a common convention.

**Comment by jmockler** [comment]
Agree.. that no-reply is a common convention. I would be ok with showing that the email is from a Maqetta server instead of 'noreply'.

**Comment by JonFerraiolo** [comment]
Sorry about piling on to a long discussion, but since we know the author's email, if we put reply-to to the author's email, then a reply email would actually get to the author, right?

**Comment by peller** [comment]
yes, sounds like that should work.  The other option is to have some sort of redirect mechanism on our servers so we would generate a unique address at the maqetta server, then deliver the e-mail to the user, sort of like github is doing right now.

**Comment by jmockler** [comment]
yes.. that would be ideal. But it sounds like @peller is saying there is a technical issue with doing that? Am I misreading @peller's last comment?

**Comment by JonFerraiolo** [comment]
@aerwin - if this is out of your technical knowledge area, perhaps you can reassign to @jhpedemonte (or at least partner with him) because he is likely to have been forced to learn send-mail things in past several months

**Comment by aerwin** [comment]
@JonFerraiolo @jhpedemonte 

-> M8

**Comment by jhpedemonte** [comment]
> ... since we know the author's email, if we put reply-to to the author's email, then a reply email would actually get to the author, right?

We don't want to do this.  It would mean that replies would go directly to the reviewer/commenter and not to Maqetta, breaking the discussion thread within Maqetta.

> ... redirect mechanism on our servers so we would generate a unique address at the maqetta server, then deliver the e-mail to the user ...

Something like this is the correct solution: with Maqetta acting as an intermediary, a user can respond to a review comment by replying to the email, which goes to Maqetta, adds the reply to the discussion thread within R&C and sends an email reply to the original commenter.

Therefore, a user can respond to a review comment in two ways:
1. from within Maqetta
2. by replying to the email notification

In both cases, a user should be able to read the full discussion within Maqetta.

**Comment by JonFerraiolo** [comment]
I agree with @jhpedemonte

**Comment by aerwin** [comment]
So, then we should close this? And, maybe open a brand new feature request for Maqetta to act as an e-mail intermediary?
